rmw_opendds_shared_cpp/src/node.cpp

```diff
@@ -310,6 +311,9 @@ create_node(
   } catch (...) {}
 
   clean_node(dpf, node);
+  if (buf) {
+    rmw_free(buf);
+  }
```

Comment:
Standard `free` will take a null pointer, so I doubt checking will be needed.

Reply:
http://docs.ros2.org/dashing/api/rcutils/allocator_8h.html

Reply:
> The default allocator uses malloc(), free(), calloc(), and realloc().

Okay, so yeah, it should be fine with a null pointer unless something else says otherwise.